Pager/PagerService.php

```diff
@@ -204,9 +204,24 @@ public function isPaginable()
     public function getPages()
     {
         $pages = $this->getMaxPages();
-        $tmp = $this->page - floor($pages / 2);
-        $begin = $tmp > $this->getFirstPage() ? $tmp : $this->getFirstPage();
-        $end = min($begin + $pages - 1, $this->getLastPage());
+
+        $tmpBegin = $this->page - floor($pages / 2);
+        $tmpEnd = $tmpBegin + $pages - 1;
+
+        if ($tmpBegin < $this->getFirstPage()) {
+            $overleft = $this->getFirstPage() - $tmpBegin;
+            $tmpEnd += $overleft;
```

Comment:
If a varable is used only one time I would suggest not to create. f.e. here `$overleft` is used only on the line below, so it would be better to write just `$tmpEnd += $this->getFirstPage() - $tmpBegin;`